### ISSUE TITLE
Players database with team association

### ISSUE THREAD

**Issue by FanaHOVA** [nickb1080/nba#17]
I was wondering if there's some way to connect the teams to the players? I'd like to do some stats study by team to figure out the best rotations, but right now the players.json only has first and last name, no position or team. Is there any endpoint we can take advantage of to do that?

**Comment by nickb1080** [comment]
@FanaHOVA Sure! `nba.stats.commonTeamRoster` is probably what you're looking for. 
For example, heres the result of `nba.stats.commonTeamRoster({teamId: 1610612744})` (Warriors)

```json
{ commonTeamRoster:
   [ { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Ognjen Kuzmic',
       num: '1',
       position: 'C',
       height: '7-0',
       weight: '260',
       birthDate: 'MAY 16, 1990',
       age: 25,
       exp: '1',
       school: null,
       playerId: 203136 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Brandon Rush',
       num: '4',
       position: 'G-F',
       height: '6-6',
       weight: '220',
       birthDate: 'JUL 07, 1985',
       age: 29,
       exp: '6',
       school: 'Kansas',
       playerId: 201575 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Marreese Speights',
       num: '5',
       position: 'C-F',
       height: '6-10',
       weight: '255',
       birthDate: 'AUG 04, 1987',
       age: 27,
       exp: '6',
       school: 'Florida',
       playerId: 201578 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Justin Holiday',
       num: '7',
       position: 'G',
       height: '6-6',
       weight: '185',
       birthDate: 'APR 05, 1989',
       age: 26,
       exp: '1',
       school: 'Washington',
       playerId: 203200 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Andre Iguodala',
       num: '9',
       position: 'G-F',
       height: '6-6',
       weight: '215',
       birthDate: 'JAN 28, 1984',
       age: 31,
       exp: '10',
       school: 'Arizona',
       playerId: 2738 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'David Lee',
       num: '10',
       position: 'F-C',
       height: '6-9',
       weight: '245',
       birthDate: 'APR 29, 1983',
       age: 32,
       exp: '9',
       school: 'Florida',
       playerId: 101135 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Klay Thompson',
       num: '11',
       position: 'G',
       height: '6-7',
       weight: '215',
       birthDate: 'FEB 08, 1990',
       age: 25,
       exp: '3',
       school: 'Washington State',
       playerId: 202691 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Andrew Bogut',
       num: '12',
       position: 'C',
       height: '7-0',
       weight: '260',
       birthDate: 'NOV 28, 1984',
       age: 30,
       exp: '9',
       school: 'Utah',
       playerId: 101106 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Leandro Barbosa',
       num: '19',
       position: 'G',
       height: '6-3',
       weight: '194',
       birthDate: 'NOV 28, 1982',
       age: 32,
       exp: '11',
       school: 'Sau Paulo, Brazil',
       playerId: 2571 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'James Michael McAdoo',
       num: '20',
       position: 'F',
       height: '6-9',
       weight: '230',
       birthDate: 'JAN 04, 1993',
       age: 22,
       exp: 'R',
       school: 'North Carolina',
       playerId: 203949 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Draymond Green',
       num: '23',
       position: 'F',
       height: '6-7',
       weight: '230',
       birthDate: 'MAR 04, 1990',
       age: 25,
       exp: '2',
       school: 'Michigan State',
       playerId: 203110 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Stephen Curry',
       num: '30',
       position: 'G',
       height: '6-3',
       weight: '190',
       birthDate: 'MAR 14, 1988',
       age: 27,
       exp: '5',
       school: 'Davidson',
       playerId: 201939 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Festus Ezeli',
       num: '31',
       position: 'C',
       height: '6-11',
       weight: '265',
       birthDate: 'OCT 21, 1989',
       age: 25,
       exp: '2',
       school: 'Vanderbilt',
       playerId: 203105 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Shaun Livingston',
       num: '34',
       position: 'G',
       height: '6-7',
       weight: '192',
       birthDate: 'SEP 11, 1985',
       age: 29,
       exp: '9',
       school: 'Peoria Central HS (IL)',
       playerId: 2733 },
     { teamID: 1610612744,
       season: '2014',
       leagueID: '00',
       player: 'Harrison Barnes',
       num: '40',
       position: 'F',
       height: '6-8',
       weight: '225',
       birthDate: 'MAY 30, 1992',
       age: 23,
       exp: '2',
       school: 'North Carolina',
       playerId: 203084 } ],
  coaches:
   [ { teamId: 1610612744,
       season: '2014',
       coachId: 'KER331857',
       firstName: 'Steve',
       lastName: 'Kerr',
       coachName: 'Steve Kerr',
       coachCode: 'steve_kerr',
       isAssistant: 1,
       coachType: 'Head Coach',
       school: 'College - Arizona',
       sortSequence: null },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'ADA652935',
       firstName: 'Ron',
       lastName: 'Adams',
       coachName: 'Ron Adams',
       coachCode: 'ron_adams',
       isAssistant: 2,
       coachType: 'Assistant Coach',
       school: 'College - Fresno Pacific',
       sortSequence: 90 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'WAL794301',
       firstName: 'Luke',
       lastName: 'Walton',
       coachName: 'Luke Walton',
       coachCode: 'luke_walton',
       isAssistant: 2,
       coachType: 'Assistant Coach',
       school: 'College - Arizona',
       sortSequence: 91 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'WAN106965',
       firstName: 'Johan',
       lastName: 'Wang',
       coachName: 'Johan Wang',
       coachCode: 'johan_wang',
       isAssistant: 3,
       coachType: 'Trainer',
       school: 'College - UC Davis',
       sortSequence: 95 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'YOD147981',
       firstName: 'Drew',
       lastName: 'Yoder',
       coachName: 'Drew Yoder',
       coachCode: 'drew_yoder',
       isAssistant: 4,
       coachType: 'Assistant Trainer',
       school: 'College - Kansas State',
       sortSequence: 96 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'COL642519',
       firstName: 'Jarron',
       lastName: 'Collins',
       coachName: 'Jarron Collins',
       coachCode: 'jarron_collins',
       isAssistant: 12,
       coachType: 'Assistant Coach for Player Development',
       school: 'College - Stanford',
       sortSequence: 92 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'FRA408656',
       firstName: 'Bruce',
       lastName: 'Fraser',
       coachName: 'Bruce Fraser',
       coachCode: 'bruce_fraser',
       isAssistant: 12,
       coachType: 'Assistant Coach for Player Development',
       school: 'College - Arizona',
       sortSequence: 93 },
     { teamId: 1610612744,
       season: '2014',
       coachId: 'LYL167481',
       firstName: 'Keke',
       lastName: 'Lyles',
       coachName: 'Keke Lyles',
       coachCode: 'keke_lyles',
       isAssistant: 14,
       coachType: 'Assistant Coach/Player Performance',
       school: 'College - Northeastern',
       sortSequence: 94 } ] }
```

**Comment by nickb1080** [comment]
`nba.stats.lineups()` landed in 1.2.0. Returns an array of objects that look like:

```json
{
  "groupId": "203084 - 201939 - 203110 - 2738 - 202691",
  "groupName": "Barnes,Harrison - Curry,Stephen - Green,Draymond - Iguodala,Andre - Thompson,Klay",
  "teamId": 1610612744,
  "teamAbbreviation": "GSW",
  "gp": 14,
  "w": 14,
  "l": 0,
  "wPct": 1,
  "min": 4.6,
  "fgm": 5,
  "fga": 7.9,
  "fgPct": 0.636,
  "fG3M": 2.5,
  "fG3A": 4,
  "fg3Pct": 0.625,
  "ftm": 3.2,
  "fta": 3.9,
  "ftPct": 0.818,
  "oreb": 0.6,
  "dreb": 3.6,
  "reb": 4.3,
  "ast": 3.6,
  "tov": 1.2,
  "stl": 1.1,
  "blk": 0.6,
  "blka": 0.2,
  "pf": 1.7,
  "pfd": 2.7,
  "pts": 15.7,
  "plusMinus": 6.5,
  "playerIds": [
    203084,
    201939,
    203110,
    2738,
    202691
  ]
}
```

**Comment by nickb1080** [comment]
`nba.stats.lineups()` landed in 1.2.0. 

The main parameter to vary is `measureType` which can take "Advanced" or "Base". Oddly, I ran it for one specific lineup and the "Base" seemed way off (at least minutes-wise).
